x-pack/plugins/aiops/public/components/document_count_content/document_count_chart/document_count_chart.tsx

```diff
@@ -0,0 +1,162 @@
+import { useAiOpsKibana } from '../../../kibana_context';
+
+export interface DocumentCountChartPoint {
+  time: number | string;
+  value: number;
+}
+
+interface Props {
+  width?: number;
+  chartPoints: DocumentCountChartPoint[];
+  timeRangeEarliest: number;
+  timeRangeLatest: number;
+  interval?: number;
+}
+
+const SPEC_ID = 'document_count';
+
+function getTimezone(uiSettings: IUiSettingsClient) {
+  if (uiSettings.isDefault('dateFormat:tz')) {
+    const detectedTimezone = moment.tz.guess();
+    if (detectedTimezone) return detectedTimezone;
+    else return moment().format('Z');
+  } else {
+    return uiSettings.get('dateFormat:tz', 'Browser');
+  }
+}
+
+export const DocumentCountChart: FC<Props> = ({
+  width,
+  chartPoints,
+  timeRangeEarliest,
+  timeRangeLatest,
+  interval,
+}) => {
+  const {
+    services: { data, uiSettings, fieldFormats, charts },
+  } = useAiOpsKibana();
+
+  const chartTheme = charts.theme.useChartsTheme();
+  const chartBaseTheme = charts.theme.useChartsBaseTheme();
+
+  const xAxisFormatter = fieldFormats.deserialize({ id: 'date' });
+  const useLegacyTimeAxis = uiSettings.get('visualization:useLegacyTimeAxis', false);
+
+  const seriesName = i18n.translate('xpack.aiops.dataGrid.field.documentCountChart.seriesLabel', {
+    defaultMessage: 'document count',
+  });
+
+  const xDomain = {
+    min: timeRangeEarliest,
+    max: timeRangeLatest,
+  };
+
+  const adjustedChartPoints = useMemo(() => {
+    // Display empty chart when no data in range
+    if (chartPoints.length < 1) return [{ time: timeRangeEarliest, value: 0 }];
+
+    // If chart has only one bucket
+    // it won't show up correctly unless we add an extra data point
+    if (chartPoints.length === 1) {
+      return [
+        ...chartPoints,
+        { time: interval ? Number(chartPoints[0].time) + interval : timeRangeEarliest, value: 0 },
+      ];
+    }
+    return chartPoints;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [chartPoints, timeRangeEarliest, timeRangeLatest, interval]);
+
+  const timefilterUpdateHandler = useCallback(
+    (ranges: { from: number; to: number }) => {
+      data.query.timefilter.timefilter.setTime({
+        from: moment(ranges.from).toISOString(),
+        to: moment(ranges.to).toISOString(),
+        mode: 'absolute',
+      });
+    },
+    [data]
+  );
+
+  const onBrushEnd = ({ x }: XYBrushEvent) => {
+    if (!x) {
+      return;
+    }
+    const [from, to] = x;
+    timefilterUpdateHandler({ from, to });
+  };
+
+  const onElementClick: ElementClickListener = ([elementData]) => {
+    const startRange = (elementData as XYChartElementEvent)[0].x;
+
+    const range = {
+      from: startRange,
+      to: startRange + interval,
+    };
+    timefilterUpdateHandler(range);
+  };
+
+  const timeZone = getTimezone(uiSettings);
+
+  return (
+    <div style={{ width: width ?? '100%' }} data-test-subj="dataVisualizerDocumentCountChart">
```

Comment:
Reference to `dataVisualizer` should be changed to `aiops`.

Reply:
Updated in fd3cbb696ab1d3bc4f6c9cbcfd9f29124e986bf1